Content.Shared/Silicons/StationAi/SharedStationAiSystem.Held.cs

```diff
@@ -107,42 +108,67 @@ private void OnMessageAttempt(BoundUserInterfaceMessageAttempt ev)
         if (ev.Actor == ev.Target)
             return;
 
+        StationAiWhitelistComponent? whitelistComponent = null;
         if (TryComp(ev.Actor, out StationAiHeldComponent? aiComp) &&
            (!ValidateAi((ev.Actor, aiComp)) ||
-            !HasComp<StationAiWhitelistComponent>(ev.Target)))
+            !TryComp(ev.Target, out whitelistComponent)))
         {
             ev.Cancel();
         }
+
+        if (whitelistComponent is { Enabled: false })
+        {
+            ev.Cancel();
+            ShowDeviceNotRespondingPopup(ev.Actor);
+        }
     }
 
     private void OnHeldInteraction(Entity<StationAiHeldComponent> ent, ref InteractionAttemptEvent args)
     {
-        // Cancel if it's not us or something with a whitelist.
-        args.Cancelled = ent.Owner != args.Target &&
-                         args.Target != null &&
-                         (!TryComp(args.Target, out StationAiWhitelistComponent? whitelist) || !whitelist.Enabled);
+        StationAiWhitelistComponent? whitelistComponent = null;
+        // Cancel if it's not us or something with a whitelist, or whitelist is disabled.
+        args.Cancelled = ent.Owner != args.Target
+                         && args.Target != null
+                         && (!TryComp(args.Target, out whitelistComponent)
+                             || !whitelistComponent.Enabled);
+        if (whitelistComponent is { Enabled: false })
+        {
+            ShowDeviceNotRespondingPopup(ent.Owner);
+        }
     }
 
     private void OnTargetVerbs(Entity<StationAiWhitelistComponent> ent, ref GetVerbsEvent<AlternativeVerb> args)
     {
-        if (!args.CanComplexInteract ||
-            !ent.Comp.Enabled ||
-            !HasComp<StationAiHeldComponent>(args.User) ||
-            !HasComp<StationAiWhitelistComponent>(args.Target))
+        if (!args.CanComplexInteract
+            || !HasComp<StationAiHeldComponent>(args.User)
+            || !HasComp<StationAiWhitelistComponent>(args.Target))
```

Comment:
I think this whitelist check is unneeded?

Reply:
It is, i belive, currently overlapping with OnHeldInteraction, but i'm not 100% sure it covers every possible scenariod

Reply:
Seemed good when i tested it - so i removed condition check duplication!